packages/opentelemetry-types/src/trace/tracer.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from './span';
+import { Attributes } from './attributes';
+
+/**
+ * Options needed for span creation
+ *
+ * @todo: Move into module of its own
+ */
+export interface SpanOptions {
+  attributes: Attributes[];
+}
+
+export interface Tracer {
+  /**
+   * Returns the current Span from the current context if available.
+   *
+   * If there is no Span associated with the current context, a default Span
+   * with invalid SpanContext is returned.
+   *
+   * @returns Span
+   */
+  getCurrentSpan(): Span;
+
+  /**
+   *
+   * @param name
+   * @param options
+   */
+  start(name: string, options: SpanOptions): Span;
+
+  /**
+   * Executes the function given by fn within the context provided by Span
+   *
+   * @param span The span that provides the context
+   * @param operation The name of the function
+   * @param fn The function to be eexcuted inside the provided context
+   */
+  withSpan<T extends (...args: unknown[]) => unknown>(
+    span: Span,
+    operation: string,
```

Comment:
Why pass the operation name again here?

Reply:
Where else did I pass in the operation name?

Reply:
What I meant is that the span already contains the operation name, so this one is redundant.

Reply:
ok removed